runtime/lua/vim/fs.lua

```diff
@@ -47,19 +48,20 @@ function M.dirname(file)
     return nil
   end
   vim.validate({ file = { file, 's' } })
-  if iswin and file:match('^%w:[\\/]?$') then
-    return (file:gsub('\\', '/'))
-  elseif not file:match('[\\/]') then
+  file = file:gsub(os_sep, '/') --[[@as string]]
```

Comment:
Worth gating this with an `if iswin` (or `if os_sep ~= '/'`)? For a long path, could potentially be wasteful on Unix.

Reply:
yeah sounds good! implemented